src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -27,6 +28,21 @@ class CertificateService(CRUDService):
         options['extend'] = 'certificate.cert_extend'
         return await self.middleware.call('datastore.query', 'system.certificate', filters, options)
 
+    @accepts(
+        Str('cert_certificate')
+    )
+    async def get_certificate_fingerprint(self, cert_certificate):
```

Comment:
Not sure it makes sense to have the certificate as argument.

How about the id? Then you can move all the logic to get the certificate from system.
It will also git the CRUD logic to get this as an item method automatically via REST (e.g. /certificate/id/{id}/fingerprint)

Also what do you think about just naming it `get_fingerprint`?
`certificate.get_certificate_fingerprint` seems a bit redundant.